### PR TITLE
REF/TST: collect astype tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_dtypes.py [hunk 1]
```diff
@@ -249,27 +249,6 @@ def test_str_to_small_float_conversion_type(self):
         expected = pd.DataFrame(col_data, columns=["A"], dtype=float)
         tm.assert_frame_equal(result, expected)
 
-    @pytest.mark.parametrize(
-        "convert_integer, expected", [(False, np.dtype("int32")), (True, "Int32")]
```
Comment (jreback) [inline pandas/tests/frame/test_dtypes.py:253]: and there are no more astype's in this file? at all? i find it hard to believe we have exactly 1 test for astype on a frame

File: pandas/tests/series/test_dtypes.py [hunk 2]
```diff
@@ -1,123 +1,21 @@
-from datetime import datetime, timedelta
-from importlib import reload
```
Comment (jreback) [inline pandas/tests/series/test_dtypes.py:2]: what exactly is left over here? i am not sure i get why you are moving to a new file than just renaming and moving out anything that isn't astype (which is a very small number)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_dtypes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_dtypes.py:253]: My bad, poorly titled PR.

DataFrame.astype tests are already in test_astype.  This is our lone test for `DataFrame.convert_dtypes` being moved to test_convert_dtypes

File: pandas/tests/series/test_dtypes.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_dtypes.py:2]: 9 tests left here for which i havent found a home, but it definitely isnt test_astype.  one probably belongs in the DataFrame tests, one for reindex, one for constructors...

i do expect this file to be killed off before too long

Comment (jreback) [reply pandas/tests/series/test_dtypes.py:2]: ok its fine, just though would be easier to rename
but ok

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
hmm

i think this one is actually failing some CI (that IS related)

e.g. CI/web and docs is failing

**Comment by jreback** [comment]
can you rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Pull request merged**